src/services/connextService.ts

```diff
@@ -0,0 +1,161 @@
+    return callDataForMeantTarget;
+  }
+
+  async prepareSwapAndXCallHelper(swapAndXCallParams: SwapAndXCallParams, signerAddress: string) {
+    const txRequest = await prepareSwapAndXCall(swapAndXCallParams, signerAddress);
+    return txRequest;
+  }
+
+  async getEstimateAmountReceived(
+    originDomain: string,
+    destinationDomain: string,
+    originToken: string,
+    amount: number
+  ) {
+    const domainConfig: { [domainId: string]: { providers: string[] } } = {};
+
+    const domainChainIds = Object.entries(SUPPORTED_CHAINS_BY_CONNEXT)
+      .filter(([key]) => typeof key === 'number')
+      .map(([key, value]) => ({ domainId: value.domainId, chainId: key }));
+
+    domainChainIds.forEach((obj) => {
+      domainConfig[obj.domainId] = { providers: [this.getRPCURL(parseInt(obj.chainId, 10))] };
+    });
+
+    const sdkConfig: SdkConfig = {
+      signerAddress: this.walletService.getAccount(),
+      network: 'mainnet', // can change it to testnet as well
+      chains: domainConfig,
+    };
+    const { sdkBase } = await create(sdkConfig);
+    const estimateReceived = await sdkBase.calculateAmountReceived(
+      originDomain,
+      destinationDomain,
+      originToken,
+      amount
+    );
+    return estimateReceived;
+  }
+
+  getDomainID(networkName: number): string {
+    const domainID: DomainID = {
+      1: '6648936',
+      137: '1886350457',
+      10: '1869640809',
+      42161: '1634886255',
+      100: '6778479',
+      56: '6450786',
+    };
+
+    return domainID[networkName];
+  }
+
+  getNativeUSDCAddress(networkName: number) {
+    const USDC_ADDRESS: DomainID = {
+      1: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
+      137: '0x2791bca1f2de4661ed88a30c99a7a9449aa84174',
+      10: '0x7F5c764cBc14f9669B88837ca1490cCa17c31607',
+      42161: '0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8',
+      100: '0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83',
+      56: '0x8AC76a51cc950d9822D68b83fE1Ad97B32Cd580d',
+    };
+    return USDC_ADDRESS[networkName];
+  }
+
+  async getTransferStatus(transactionHash: string) {
+    try {
+      const domainConfig: { [domainId: string]: { providers: string[] } } = {};
+
+      const domainChainIds = Object.entries(SUPPORTED_CHAINS_BY_CONNEXT)
+        .filter(([key]) => typeof key === 'number')
+        .map(([key, value]) => ({ domainId: value.domainId, chainId: key }));
+
+      domainChainIds.forEach((obj) => {
+        domainConfig[obj.domainId] = { providers: [this.getRPCURL(parseInt(obj.chainId, 10))] };
+      });
+
+      const sdkConfig: SdkConfig = {
+        signerAddress: this.walletService.getAccount(),
+        network: 'mainnet', // can change it to testnet as well
+        chains: domainConfig,
+      };
+      const { sdkUtils } = await create(sdkConfig);
+      const params: { transactionHash: string } = {
+        transactionHash,
+      };
+      const transferStatus = await sdkUtils.getTransfers(params);
```

Comment:
how can it know in which chain is it bridging if it only has the transactionHash?

Reply:
So it is the internal function we use for connextscan, it can fetch the whole transaction from origin transactionHash itself, it will give you chains and other details as well